java/maven/src/org/netbeans/modules/maven/execute/MavenCommandLineExecutor.java

```diff
@@ -445,6 +445,18 @@ private static List<String> createMavenExecutionCommand(RunConfig config, Constr
                 LOGGER.log(Level.FINE, "Could not canonicalize " + basedir, x);
             }
         }
+    
+        for (Map.Entry<? extends String, ? extends String> entry : config.getOptions().entrySet()) {
+            String key = entry.getKey();
+            String value = quote2apos(entry.getValue());
+            if (MavenCommandLineOptions.optionRequiresValue(key) && (value.isBlank() || value.equals("${" + key + "}"))) { //NOI18N
```

Comment:
why the check for `${optionName}` ?
Consider to print a warning visibly to the user: if trying to pass no value for an option that is known to have one, it's a bug somewhere in the code.

Maybe I am picky, but `isBlank()` is true also for "  " (two spaces); I wouldn't equal that to a no-value. Consider value quoting (see proprerties below).